gc/base/MemorySubSpace.cpp

```diff
@@ -919,16 +919,23 @@ MM_MemorySubSpace::systemGarbageCollect(MM_EnvironmentBase* env, uint32_t gcCode
 	}
 
 	if (_collector && _usesGlobalCollector) {
-		/* TODO: This is bogus for multiple memory spaces - should ask the space, not the heap */
-		_extensions->heap->getResizeStats()->setFreeBytesAtSystemGCStart(getApproximateActiveFreeMemorySize());
-
 		env->acquireExclusiveVMAccessForGC(_collector);
-		reportSystemGCStart(env, gcCode);
-
-		/* system GCs are accounted into "user" time in GC/total time ratio calculation */
-		_collector->garbageCollect(env, this, NULL, gcCode, NULL, NULL, NULL);
-
-		reportSystemGCEnd(env);
+		#if defined(OMR_GC_IDLE_HEAP_MANAGER)
+		if (isSystemGCAllowed(env, gcCode)) {
+		#endif
+			/* TODO: This is bogus for multiple memory spaces - should ask the space, not the heap */
+			_extensions->heap->getResizeStats()->setFreeBytesAtSystemGCStart(getApproximateActiveFreeMemorySize());
```

Comment:
I understand that is moved down so that we do not take this snapshot if we do not end up doing the system GC.  bigger concern is that 'is allowed' logic is AFTER acquiring exclusive access - seems like a wasted exercise if sysGC is not performed. Could we invoke 'is allowed' logice prior to invoking top level sysGC? 

I did not look into the rest of the code yet, but what happens if we sysGC is not 'allowed' - do we still take exclusive access to walk&decommit Tenure free pool?